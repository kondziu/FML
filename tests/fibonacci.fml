function fibonacci(value) ->
         if value <  0              then print("Error: value must be a Natural number.\n")
    else if value == 0              then 0
    else if value == 1 | value == 2 then 1
    else                                 fibonacci(value - 1) + fibonacci(value - 2);

let i = -1;
while i < 16 do
begin
  print("~ => ~\n", i, fibonacci(i));
  i <- i + 1;
end

// > Error: value must be a Natural number.
// > -1 => null
// > 0 => 0
// > 1 => 1
// > 2 => 1
// > 3 => 2
// > 4 => 3
// > 5 => 5
// > 6 => 8
// > 7 => 13
// > 8 => 21
// > 9 => 34
// > 10 => 55
// > 11 => 89
// > 12 => 144
// > 13 => 233
// > 14 => 377
// > 15 => 610