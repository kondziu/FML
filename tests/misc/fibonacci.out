Error: value must be a Natural number.
-1 => null
0 => 0
1 => 1
2 => 1
3 => 2
4 => 3
5 => 5
6 => 8
7 => 13
8 => 21
9 => 34
10 => 55
11 => 89
12 => 144
13 => 233
14 => 377
15 => 610
